kof-operator/internal/k8s/pods.go

```diff
@@ -0,0 +1,22 @@
+package k8s
+
+import (
+	"context"
+
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func GetCollectorPods(ctx context.Context, k8sClient client.Client) (*corev1.PodList, error) {
+	podList := &corev1.PodList{}
+
+	if err := k8sClient.List(
+		ctx,
+		podList,
+		client.MatchingLabels(map[string]string{"app.kubernetes.io/component": "opentelemetry-collector"}),
```

Comment:
Let's add a [podAnnotation](https://github.com/open-telemetry/opentelemetry-operator/blob/main/docs/api/opentelemetrycollectors.md#opentelemetrycollectorspec) to the collector that has prometheus receiver to additionally filter out collectors without receiver to avoid error noise in logs.
```
k0rdent.mirantis.cof/kof-prometheus-receiver: true
```